tensorflow/java/src/main/java/org/tensorflow/Operation.java

```diff
@@ -97,6 +99,46 @@ public Output output(int idx) {
     return new Output(this, idx);
   }
 
+  @Override
+  public int hashCode() {
+    Graph.Reference r = graph.ref();
+    try {
+      // xor the first and last 32 bits of the underlying 64-bit pointer value.
+      return (int) (unsafeNativeHandle ^ (unsafeNativeHandle >>> 32));
```

Comment:
Do we need to grab a reference on the graph? Seems like the contract for `hashCode` is such that it would be okay to write this as:

```java
public int hashCode() { return Long.valuOf(unsafeNativeHandle).hashCode(); }
```
?